### DISCUSSION TITLE
PHP Spark Migrate Issue, database: Connection refused

### DISCUSSION THREAD

**Discussion by bribar** [codeigniter4/shield#306] [Q&A]
System Environment:
macOS Monterey 12.4
MAMP Pro 6.6
- PHP 7.4.21
- Apache 2.4.46
- MySql 5.7.34

Codeigniter is latest version

I've seen the other issues posted in regards to Sqlite and database connection issues, but they don't seem to fix.

I followed the appstarter instructions here https://codeigniter4.github.io/userguide/installation/installing_composer.html

Codeigniter:
Updated Config\App.php base url
Updated Config\Database.php settings for my DB (not using env file)
Checked Config\Logger and threshold set to 4

Through the Home Controller (Browser) I can successfully connect to DB and make a simple query that returns results so I know my database settings are valid

When I proceed to install Shield following these instructions https://github.com/codeigniter4/shield/blob/develop/docs/install.md and running "php spark migrate" (tried the -all option too, also I've removed the _support tests like suggested.) it throws two errors depending on the settings below...

Using these settings:

```php
public $default = [
        'DSN'      => '',
        'hostname' => 'localhost',
        'username' => 'root',
        'password' => 'root',
        'database' => 'btb',
        'DBDriver' => 'MySQLi',
        'DBPrefix' => '',
        'pConnect' => false,
        'DBDebug'  => (ENVIRONMENT !== 'production'),
        'charset'  => 'utf8',
        'DBCollat' => 'utf8_general_ci',
        'swapPre'  => '',
        'encrypt'  => false,
        'compress' => false,
        'strictOn' => false,
        'failover' => [],
        'port'     => 3306,
    ];
```

Running the command logs "ERROR - 2022-07-20 19:11:46 --> Error connecting to the database: No such file or directory"

Only changing localhost => 127.0.0.1

Running the command logs "ERROR - 2022-07-20 19:15:05 --> Error connecting to the database: Connection refused"

Doesn't tell me why it's refused...

I've only made the necessary changes to settings to make it run. I haven't moved folders, like the public folder, that some have suggested might be the issue.

Also running "php spark shield:setup" spits out - Command "shield:setup" not found. Probably cause previous commands are failing?

It's a very base install to test Shield. Pretty sure all my system requirements are adequate. Not sure what else to do.

Folder structure is:
ci4 folder -> appstarter folder -> framework folders (app,public,vendor, etc)

Did I miss a solution for my case? Can I manually create necessary DB tables for Shield and skip the migration and do the Manual Setup?

**Comment by kenjis** [comment]
Is your MySQL running?

**Reply by bribar** [reply]
Yes, through MAMP.
 
Do I have to run it through spark?

**Reply by kenjis** [reply]
No.

**Reply by kenjis** [reply]
Your MySQL port number is 3306 (default)?

**Reply by bribar** [reply]
Yes

**Comment by kenjis** [comment]
Run:
```console
$ php --ini
```
Is the `php` you are using MAMP's php?

**Reply by bribar** [reply]
Yes
```
Configuration File (php.ini) Path: /usr/local/etc/php/7.4
Loaded Configuration File:         /usr/local/etc/php/7.4/php.ini
Scan for additional .ini files in: /usr/local/etc/php/7.4/conf.d
Additional .ini files parsed:      /usr/local/etc/php/7.4/conf.d/ext-opcache.ini
```

**Reply by kenjis** [reply]
MAMP's `php.ini` is installed in `/Applications/MAMP/bin/php/php*.*.*/conf/`.

**Reply by bribar** [reply]
Ok I see it there. Are you asking me to do something?

**Reply by kenjis** [reply]
You are not using MAMP's `php` command, so it can't access MySQL.

**Reply by bribar** [reply]
That's why the php spark command won't work?

**Reply by kenjis** [reply]
Yes.

**Reply by bribar** [reply]
Ok. Is there a setting I can set in the project to make Spark look to the MAMP environment? Or run it differently?

Sorry for many questions, these aren't my strengths.

**Reply by bribar** [reply]
I guess the tables are here...

2020-12-28-223112_create_auth_tables.php

I could just manually run these? And then Manually run 1-3 in the setup?

And should be good?

**Reply by kenjis** [reply]
Your problem is you don't run MAMP's `php`.

> I could just manually run these?

No. You cannot run the php file directly. Because the framework is not built like that way.
You must run always the framework command file. In this case `spark`.

**Comment by kenjis** [comment]
Try:
```console
$ /Applications/MAMP/bin/php/php7.4.21/php spark shield:setup
```

**Reply by bribar** [reply]
It doesn't work
`zsh: no such file or directory: /Applications/MAMP/bin/php/php7.4.21/php`
I've tried `php spark routes` and that works

**Reply by kenjis** [reply]
See *Activate command line shortcuts for the selected PHP version, pear & pecl*
in https://documentation.mamp.info/en/MAMP-PRO-Mac/Languages/PHP/index.html
